app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/DisplayImageFragment.kt

```diff
@@ -0,0 +1,117 @@
+import android.view.ViewGroup
+import android.widget.ImageButton
+import android.widget.TextView
+import android.widget.Toast
+import androidx.activity.OnBackPressedCallback
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import com.github.HumanLearning2021.HumanLearningApp.R
+import com.github.HumanLearning2021.HumanLearningApp.databinding.FragmentDisplayImageBinding
+import com.github.HumanLearning2021.HumanLearningApp.databinding.FragmentLearningBinding
+import com.github.HumanLearning2021.HumanLearningApp.hilt.Demo2Database
+import com.github.HumanLearning2021.HumanLearningApp.hilt.ScratchDatabase
+import com.github.HumanLearning2021.HumanLearningApp.model.CategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.model.Category
+import com.github.HumanLearning2021.HumanLearningApp.model.DatabaseManagement
+import com.github.HumanLearning2021.HumanLearningApp.view.learning.LearningFragmentArgs
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class DisplayImageFragment : Fragment() {
+    private lateinit var parentActivity: FragmentActivity
+
+    @Inject
+    @Demo2Database
+    lateinit var dbManagement: DatabaseManagement
+
+    private var picture: CategorizedPicture? = null
+    private lateinit var datasetId: String
+    private lateinit var category: Category
+
+    private val args: DisplayImageFragmentArgs by navArgs()
+    private var _binding: FragmentDisplayImageBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        parentActivity = requireActivity()
+        _binding = FragmentDisplayImageBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+
+        picture = args.picture
+        datasetId = args.datasetId
+        category = picture!!.category
+        parentActivity.findViewById<TextView>(R.id.display_image_viewCategory).text = category.name
+        picture!!.displayOn(
+            parentActivity,
+            parentActivity.findViewById(R.id.display_image_viewImage)
+        )
+
+        parentActivity.findViewById<ImageButton>(R.id.display_image_delete_button)
+            .setOnClickListener {
+                removePicture()
+            }
+
+        val callback = object : OnBackPressedCallback(true){
+            override fun handleOnBackPressed() {
+                findNavController().popBackStack()
+            }
+        }
+
+        requireActivity().onBackPressedDispatcher.addCallback(callback)
+
+    }
+
```

Comment:
You deleted the setRepresentativePicture function, I know this function was not implemented yet but I think it's is better to have it empty with a TODO so that we don't forget to it :)

Reply:
done